requirements.txt

```diff
@@ -1,32 +1,31 @@
-alembic==1.13.1
+alembic==1.14.1
 annotated-types==0.6.0
 anyio==4.3.0
-blinker==1.7.0
-certifi==2024.2.2
-click==8.1.7
+blinker==1.9.0
+certifi==2025.1.31
+click==8.1.8
 distro==1.9.0
-Flask==3.0.2
-Flask-Migrate==4.0.5
+Flask==3.1.0
+Flask-Migrate==4.1.0
 Flask-SQLAlchemy==3.1.1
-greenlet==3.0.3
 h11==0.14.0
 httpcore==1.0.5
 httpx==0.27.0
-idna==3.7
+idna==3.10
 iniconfig==2.0.0
-itsdangerous==2.1.2
-Jinja2==3.1.3
-Mako==1.3.2
-MarkupSafe==2.1.5
-packaging==23.2
-pluggy==1.4.0
-psycopg2-binary==2.9.9
+itsdangerous==2.2.0
+Jinja2==3.1.5
+Mako==1.3.9
+MarkupSafe==3.0.2
+packaging==24.2
+pluggy==1.5.0
+psycopg2-binary==2.9.10
 pydantic==2.7.1
 pydantic_core==2.18.2
```

Comment:
Any idea why this stuff is here? It [sounds like](https://pypi.org/project/pydantic/) this is related to doing type conversions in db models, but I don't see it being used.

I removed these and I was still able to setup and run the repo as is, though I don't know whether these would come into play in the work students do. There are a few other dependencies in here that I don't recognize that might be related to these?

This popped out for me because pip was trying to build this from source and was taking a long time.